File: solutions/1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {

    const res: number[] = [];

    for(let i = 0; i < nums.length; i += 1) {
        for(let j = i + 1; j < nums.length; j += 1) {
            if(nums[i] + nums[j] === target) {
                res[0] = i;
                res[1] = j;
                break;
            }
        }
    }

    return res;
    
};

File: solutions/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        num_to_idx = {}

        for idx, num in enumerate(nums):
            difference = target - num
            if difference in num_to_idx:
                return [num_to_idx[difference], idx]
            num_to_idx[num] = idx